warehouse/utils/security_policy.py

```diff
@@ -0,0 +1,95 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pyramid.authorization import Authenticated, Everyone
+from pyramid.interfaces import ISecurityPolicy
+from zope.interface import implementer
+
+from warehouse.accounts.interfaces import IUserService
+
+
+class SecurityPolicy:
+    def __init__(self, callback=None):
+        self._callback = callback
+
+    def identity(self, request):
```

Comment:
N.B. This is currently a generic `identity` for when policies always boil down to user identities. I'll probably need to make it less generic or remote it entirely when we allow for project identities.

Reply:
Removed.